lib/core/src/lib/form/components/widgets/core/form-field-validator.ts

```diff
@@ -583,3 +584,39 @@ export const FORM_FIELD_VALIDATORS = [
     new MinDateTimeFieldValidator(),
     new MaxDateTimeFieldValidator()
 ];
+
+export const momentToDateFnsMap = {
+    M: 'M',
+    D: 'd',
+    Y: 'y',
+    A: 'a'
+};
+
+export const dateFnsToMomentMap = {
+    M: 'M',
+    d: 'D',
+    y: 'Y',
+    a: 'A'
+};
+
+export function convertMomentToDateFnsFormat(dateDisplayFormat: string): string {
+    for (const [search, replace] of Object.entries(momentToDateFnsMap)) {
+        dateDisplayFormat = dateDisplayFormat.replace(new RegExp(search, 'g'), replace);
+    }
+    return dateDisplayFormat;
+}
+
+export function convertDateFnsToMomentFormat(dateDisplayFormat: string): string {
```

Comment:
move all these functions to `DateFnsUtils` class to have a single place for all date/time code

Reply:
moved all logic to DateFnsUtils